hknweb/exams/views.py

```diff
@@ -12,13 +14,40 @@ def index(request):
 	return render(request, 'exams/index.html', context)
 
 def exams_for_course(request, department, number):
-	department = Department.objects.filter(name__exact=department).get()
+
+	department = Department.objects.filter(abbreviated_name__exact=department).get()
 	course = Course.objects.filter(department__exact=department.id).filter(number__exact=number).get()
 	semesters = CourseSemester.objects.filter(course__exact=course.id)
 
+	print(semesters)
+
 	context = {
 		'course': course,
 		'semesters': semesters
 	}
 
-	return render(request, 'exams/exams-course.html', context)
\ No newline at end of file
+	return render(request, 'exams/exams-course.html', context)
+
+
+def add_exam(request):
+	if request.method == 'POST':
+		form = ExamUploadForm(request.POST, request.FILES)
+		if form.is_valid():
+			semester = form.cleaned_data['semester']
+			course = form.cleaned_data['course']
+			exam = form.cleaned_data['exam']
+			type = form.cleaned_data['type']
+
+			courseSemester = CourseSemester.objects.filter(course=course, semester=semester).first()
```

Comment:
I think we actually want to throw an error with `get()` if we manage to get two coursesemesters with the same course and semester.

Reply:
What should happen if an error is thrown?

Reply:
Done
Returns HttpResponseBadRequest with Error Code 409 and message